test/functional/wallet_txn_doublespend.py

```diff
@@ -34,6 +34,14 @@ def setup_network(self):
     def run_test(self):
         # All nodes should start with 1,250 BTC:
         starting_balance = 1250
+
+        # All nodes should be out of IBD.
+        # If the nodes are not all out of IBD, that can interfere with
+        # blockchain sync later in the test when nodes are connected, due to
+        # timing issues.
+        for n in self.nodes:
+            assert n.getblockchaininfo()["initialblockdownload"] == False
```

Comment:
Couldn't this be moved to `setup_nodes` under a condition of `if not self.setup_clean_chain`. This way there'd be a nice assumption that all nodes in tests could send and receive txs right away initially.

If the assertion failed, you could put a comment like `# Cache might be outdated, please run test through the test_runner script`